### PR TITLE
Add new Host Metrics dashboard

### PR DESCRIPTION
**Pull request by gdfast**
What does this PR do?
---------------------

- Adds a new Host Metrics dashboard under collector-dashboards/otel-collector-host-metrics-dashboard and adds the dashboard to README.md, main.tf, and outputs.tf
- Fixes some typos

For what purpose?
---------------------

The main purpose of this change is to add a new Host Metrics dashboard. It's essentially the OTel-native hostmetricsreceiver equivalent of the Host Metrics (Prometheus) dashboard. Adding it here will allow us to make it available in the prebuilt dashboard library.

## PR checklist

Please confirm the following items:
- [x] At least one screenshot of the proposed dashboard is included in this PR. If you're proposing substantive changes to queries or new queries then please ensure your screenshot shows relevant data.
- [x] All chart names are in [Title case](https://en.wikipedia.org/wiki/Title_case).
- [x] All query names are lower case letters, beginning the first query of each chart as "a" and proceeding alphabetically ... "b", "c", etcetera.

Screenshot 2023-10-05 at 4 19 35 PM

### CODE REVIEW

File: collector-dashboards/otel-collector-host-metrics-dashboard/main.tf [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+      query {
+        query_name   = "a"
+        display      = "line"
+        hidden       = false
+        query_string = "metric system.network.io | filter (((\"k8s.node.name\" == $k8s_node) && (\"k8s.cluster.name\" == $k8s_cluster)) && (\"direction\" == \"transmit\")) | rate | group_by [\"device\"], sum"
+      }
+    }
+
+    text_panel {
```
Comment (nslaughter) [inline collector-dashboards/otel-collector-host-metrics-dashboard/main.tf:205]: **nit:** to me it's simpler to read this config if the descriptive metadata comes first. It's not a requested change unless we're just falling into convention or default we could change elsewhere.

File: outputs.tf [hunk 2]
```diff
@@ -114,6 +114,11 @@ output "lightstep_otel_collector_host_dashboard_url" {
   description = "Cloud Observability OpenTelemetry / Host Dashboard"
 }
 
+output "lightstep_otel_collector_host_metrics_dashboard_url" {
+  value       = module.lightstep_otel_collector_host_metrics_dashboard.dashboard_url
+  description = "Cloud Observability OpenTelemetry Host Metrics  Dashboard"
```
Comment (nslaughter) [inline outputs.tf:119]:
```suggestion
  description = "Cloud Observability OpenTelemetry Host Metrics Dashboard"
```

Comment (gdfast) [reply outputs.tf:119]: Ah thanks. This blob is actually auto-generated, but I found that I had a trailing space in the dashboard title

Comment (nslaughter) [reply outputs.tf]: Sounds like you got it. I'm going to mark resolved.

File: collector-dashboards/otel-collector-host-metrics-dashboard/main.tf (hunk 1 above)
Comment (gdfast) [reply collector-dashboards/otel-collector-host-metrics-dashboard/main.tf:205]: I tend to agree, and I'll make the change. The thing is though, I made this Terraform config by using our Terraform exporter tool, which places `text_panels` at the bottom of a group in a config file. You'll notice from the `x_pos` and `y_pos` that this text panel will show up first in the actual dashboard.

**Review by adil-sultan — APPROVED** [review]

**Pull request merged**